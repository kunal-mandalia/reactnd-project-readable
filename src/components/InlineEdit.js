import React from 'react'
import PropTypes from 'prop-types'
import '../styles/InlineEdit.css';

const InlineEdit = ({ placeholder, editMode, value, onChange, className, multiline, autoFocus }) => (
  editMode ? (
    <textarea
      placeholder={placeholder}
      className={`edit-input ${className}`}
      rows={multiline ? '4' : '1'}
      cols='50'
      value={value}
      onChange={onChange}
      autoFocus={autoFocus}
      ref={textarea => textarea && (autoFocus && textarea.focus())}
    />
  ) : (
    <div
      className={`inline-edit-read-only ${className}`}    
    >{value}</div>
  )
)

InlineEdit.defaultProps = {
  placeholder: '',
  editMode: false,
  multiline: false
}

InlineEdit.propTypes = {
  placeholder: PropTypes.string,
  editMode: PropTypes.bool.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  multiline: PropTypes.bool,
  autoFocus: PropTypes.bool
}

export default InlineEdit
