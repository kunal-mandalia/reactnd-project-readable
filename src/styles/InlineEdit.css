.inline-edit {
  margin-top: 4px;
  min-height: 20px;
  box-sizing: border-box;
  width: 100%;
}

.edit-input {
  box-sizing: border-box;
  width: 100%;
  font-size: 1em;
  font-family: inherit;
  background-color: rgba(0,0,0,0.05);
  border: none;
  border-left: solid 4px rgba(0,0,0,0);
  outline: none;
  padding: 2px 2px 2px 8px;
}

.edit-input-changed {
  border-left: solid 4px paleturquoise;
}

.inline-edit-editable {
  text-align: left;
}

.inline-edit-read-only-wrapper {
  display: flex;
}

.inline-edit-read-only {
  text-align: left;
}

.actions {
  text-align: left;
}

.link-button {
  margin: 2px 4px;
  font-size: small;
  color: grey;
  cursor: pointer;
}
.link-button:hover {
  color: black;
}

.pull-right {
  float: right;
}

.post-title {
  font-weight: bold;
  color: #545454;
}
