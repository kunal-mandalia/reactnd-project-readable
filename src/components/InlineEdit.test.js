import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import InlineEdit from './InlineEdit'

const mockOnSave = jest.fn()
const mockOnDelete = jest.fn()

const props = {
  initialValue: `I'm a post`,
  onSave: mockOnSave,
  onDelete: mockOnDelete
}

describe(`<InlineEdit .../>`, () => {
  it(`should initialise read only mode`, () => {
    const wrapper = shallow(<InlineEdit {...props} />)
    expect(wrapper.state().editMode).toBeFalsy()
  })

  it(`should change to edit mode when edit button clicked`, () => {
    const wrapper = mount(<InlineEdit {...props} />)
    wrapper.find('#edit-button').simulate('click')
    expect(wrapper.state().editMode).toBeTruthy()
  })

  it(`should update value in edit mode on save`, () => {
    const newValue = `I am updated`
    const wrapper = mount(<InlineEdit {...props} />)
    wrapper.find('#edit-button').simulate('click')
    wrapper.find('#edit-input').simulate('change', {target: {value: newValue}})
    wrapper.find('#save-button').simulate('click')
    expect(mockOnSave.mock.calls[0][0]).toBe(newValue)
  })

  it(`should call delete prop on delete`, () => {
    const wrapper = mount(<InlineEdit {...props} />)
    wrapper.find('#edit-button').simulate('click')
    wrapper.find('#delete-button').simulate('click')
    expect(mockOnDelete.mock.calls).toHaveLength(1)
  })

  it(`should match snapshot`, () => {
    const tree = renderer.create(<InlineEdit {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
