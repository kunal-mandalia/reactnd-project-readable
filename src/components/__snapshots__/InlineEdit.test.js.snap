// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineEdit .../> should match snapshot 1`] = `
<div
  className="inline-edit-read-only undefined"
/>
`;
